### ISSUE TITLE
`prefer-tacit`: Not reporting `map(x => Boolean(x))`

### ISSUE THREAD

**Issue by abrahamguo** [eslint-functional/eslint-plugin-functional#801]
## Bug Report

`prefer-tacit` does not report:
```
/* eslint functional/prefer-tacit: "error" */
const foo = [1, 2, 3].map(x => Boolean(x));
```

### Expected behavior

Suggest changing to `map(Boolean)`

### Actual behavior

No error was reported.
